src/main/java/net/dv8tion/jda/core/entities/MessageActivity.java

```diff
@@ -0,0 +1,216 @@
+/*
+ *     Copyright 2015-2018 Austin Keener & Michael Ritter & Florian Spieß
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.dv8tion.jda.core.entities;
+
+import net.dv8tion.jda.core.utils.Checks;
+
+import javax.annotation.Nullable;
+import java.time.OffsetDateTime;
+
+/**
+ * Represents a {@link net.dv8tion.jda.core.entities.Message} activity.
+ */
+public class MessageActivity
+{
+    private final ActivityType type;
+    private final String partyId;
+    private final Application application;
+
+    MessageActivity(ActivityType type, String partyId, Application application)
+    {
+        this.type = type;
+        this.partyId = partyId;
+        this.application = application;
+    }
+
+    /**
+     * The current {@link net.dv8tion.jda.core.entities.MessageActivity.ActivityType ActivityType}
+     *
+     * @return the type of the activity.
+     */
+    public ActivityType getType()
+    {
+        return type;
+    }
+
+    /**
+     * The party id discord uses internally, it may be {@code null}.
+     *
+     * @return the parties id.
+     */
+    @Nullable
+    public String getPartyId()
+    {
+        return partyId;
+    }
+
+    /**
+     * The {@link net.dv8tion.jda.core.entities.MessageActivity.Application Application} this {@link net.dv8tion.jda.core.entities.MessageActivity MessageActivity} may have.
+     *
+     * @return A possibly-null {@link net.dv8tion.jda.core.entities.MessageActivity.Application}.
+     */
+    public MessageActivity.Application getApplication()
+    {
+        return application;
+    }
+
+    /**
+     * Represents the {@link net.dv8tion.jda.core.entities.Message} application if the
+     * {@link net.dv8tion.jda.core.entities.MessageActivity.ActivityType ActivityType} of the
+     * {@link net.dv8tion.jda.core.entities.MessageActivity MessageActivity} not equals
+     * {@link net.dv8tion.jda.core.entities.MessageActivity.ActivityType#LISTENING ActivityType.LISTENING}.
```

Comment:
What does this even mean?

Reply:
Can we reduce this to `Represents the {@link ...} application of an {@link ... MessageActivity}`
From what I can tell the types don't guarantee the application to be present since you only add it when its present in the builder. And even then, this should be documented somewhere else like in the getter rather than class-level.